### DISCUSSION TITLE
Disable newline comment

### DISCUSSION THREAD

**Discussion by teebow1e** [NvChad/NvChad#2411] [Q&A] [answered]
Hi everyone! How can I disable the comment for the newline if the previous line is commented?

I have tried the following:
- option in init.lua with `vim.opt_local.formatoptions:remove({"r", "o"})`

Anyone has any idea?

**Comment by siduck** [comment, marked as answer]
![image](https://github.com/NvChad/NvChad/assets/59060246/f152aaee-127e-41bc-b5a5-4a4de165ec25)

thats the formatoptions we have, we just use the defaults, and you are trying to remove r and o from it which dont even exist

**Reply by teebow1e** [reply]
But even with the default format options, I still get commented line when press enter/o.

**Reply by skochvi** [reply]
I also had this issue. A cursory investigation suggests that r and o are set by a lot of neovim's bundled filetype plugins, so it's not an NvChad thing.
Put your code in ```custom/chadrc.lua```, rather than ```init.lua```.

**Reply by siduck** [reply]
@skochvi no, in init.lua

**Reply by skochvi** [reply]
For me, it only works when I put that line of code in chadrc.lua.
Here is what I get from ```:set formatoptions```:

Without the line:
```formatoptions=jcroql```

With the line in ```custom/init.lua```:
```formatoptions=jcroql```

With the line in ```custom/chadrc.lua```:
```formatoptions=lcjq```

I'm not sure which plugins are setting formatoptions or when, but putting the line in init.lua doesn't work, and putting it in chadrc.lua does.
(Also, thank you for putting NvChad together)

**Reply by siduck** [reply]
@skochvi  impossible, check

https://github.com/NvChad/NvChad/assets/59060246/82645f21-10b5-43ed-9696-abca06d49f82

**Reply by skochvi** [reply]
As I mentioned earlier, the problem, such as it is, seems to be with neovim's bundled filetype plugins setting ```formatoptions``` (just run ```rg formatoptions``` in ```$VIMRUNTIME```). So if the plugin runs after init.vim, then any letter that has been _removed_ from ```formatoptions``` will just be replaced again. chadrc runs after the plugin, I figure. So I'm pretty sure you'll be able to replicate this if you open a file.

**Reply by siduck** [reply]
@skochvi hunt that plugin and just override its config function & append formatoptions there

**Reply by skochvi** [reply]
Well, it seems to be working fine as it is, and the the ftplugins that are apparently setting ```formatoptions``` are very numerous, are written in vimscript, and have no config functions. Is there any reason why the the solution I described above is inadequate?

**Reply by siduck** [reply]
@skochvi  its not inadequate, chadrc is meant to be a table  that replicate strucutre of core/default_config.lua, if you keep adding un-needed stuffs there then it'd  get messy. Idk which plugin you're talking about! but if its syntax related then check if there's a tsparser available for it
